lib/Dialect/FIRRTL/Transforms/ExtractClasses.cpp

```diff
@@ -46,6 +54,111 @@ struct Property {
   Location loc;
 };
 
+/// Helper to get or create a Value from an ObjectFieldOp. This consults a cache
+/// to avoid re-creating ObjectFieldOps. An ObjectOp is looked up or created,
+/// and the field corresponding to the instance output is accessed. Note that
+/// this is able to work locally with just the instance and the OpResult, and
+/// the rest of the pass ensures the correct ClassOp is created.
+Value ExtractClassesPass::getOrCreateObjectValue(
```

Comment:
Maybe name this `getOrCreateObjectFieldValue` to indicate this is for the field of an object, not the object itself.